doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,587 @@
+
+            Regex: ^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$
+        
+        + email: `an@email.com` (string, required)
+        
+            The user has to supply a valid email address.
+        
++ Response 200 (application/json)
+    + Attributes (object)
+        + message (string) - Detailed response information, normally success message
+        + i18n (string) - A unique token representing above message; can be used for internationalization in frontends
+
+            A comprehensive list of possible messages:
+            - `USER_CREATED_SUCCESSFULLY` if the account has been created successfully
+        
++ Response 400 (application/json)
+    + Attributes (object)
+        + status (string) - Textual description of the HTTP status
+        + message (string) - Detailed error information, e.g. hint about malformed request
+        + i18n (string) - A unique token representing above error description message; can be used for internationalization in frontends
+
+            A comprehensive list of possible errors:
+            - `USER_CREATE_MISSING_USERNAME` if no input for `username` was submitted
+            - `USER_CREATE_MISSING_PASSWORD` if no input for `password` was submitted
+            - `USER_CREATE_MISSING_EMAIL` if no input for `email` was submitted
+            - `USER_CREATE_INVALID_USERNAME` if the supplied `username` does not match the given criteria (regex: ^[a-zA-Z0-9\-\.]{3,20}$)
+            - `USER_CREATE_INVALID_PASSWORD` if the supplied `password` does not satisfy the given criteria (regex: ^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$)
+            - `USER_CREATE_INVALID_EMAIL` if the supplied `email` is not a valid email
+
++ Response 422 (application/json)
+    + Attributes (object)
+        + status (string) - Textual description of the HTTP status
+        + message (string) - Detailed error information, e.g. hint about malformed request
+        + i18n (string) - A unique token representing above error description message; can be used for internationalization in frontends
+
+            A comprehensive list of possible errors:
+            - `USER_CREATE_USERNAME_DOES_ALREADY_EXIST` if the supplied `username` is already taken
+        
+        
+## Update user [PUT /{name}]
+
+This method can be used to edit an user. If no `name` is supplied, the currently authenticated user is chosen as target. By using this method it is also possible to change the rank of users. Changing the rank as well as other users in general can only be done by admins (rank 3) though.
+
++ Parameters
+    + name (string, optional) - The `username` of an user which is then taken as target for the operation
+
++ Request (application/json)
+    + Attributes (object)
+        + email: `a-new@email.com` (string, optional)
+            
+            The new email address which should be used for the user. It has to be a valid email.
+
+        + password: `a_new-secureP4ssw0rd` (string, optional)
```

Comment:
please use some random string such as "waePh0IuPohhi2HiEeShee8iohyi9Ai", we should not recommend to use such weak passwords. And suggest that the password should be long and different to the previous one.